### DISCUSSION TITLE
MIMIC III Heparin continuous infusion start and endtime

### DISCUSSION THREAD

**Discussion by heisenbug-1** [MIT-LCP/mimic-code#1610] [MIMIC-III] [answered]
Hello! I apologize in advance if this question has already been asked. 

I extracted patient data from MIMIC-IV, and it was quite straightforward.
Now, I need to extract some data from MIMIC-III, and this is where i'm stuck.
 I need all data regarding patients that received heparin some of which was administered  intravenously, especially:
 - start time of infusion
 - duration
 - total amount
 
  I'm working with the CareVue subset (that does not include records from MIMIC-IV)
 I started with
```
 SELECT * 
FROM mimiciii.inputevents_cv
WHERE itemid = 30025
ORDER BY subject_id, charttime
```
There are also other items that refer to heparin, but let's limit my search to 30025 for now.
My question is, how do I extract start and end times for infusions? In the inputevents table in MIMIC-IV there were dedicated columns for start and end timestamps, but the MIMIC_III  inputevents_cv table  only contains the charttime. Also, is there a way to group the rows that refer to the same event together?

**Comment by alistairewj** [comment, marked as answer]
This was actually why we ended up with inputevents_cv and inputevents_mv in
MIMIC-III.

In CareVue, the data are documented usually hourly. Effectively the nurse
validates the amount received. Separately, the nurse can document the rate.
You run into a lot of situations where the documented rate doesn't match
the amount, and then you read the nursing note and determine it was because
the heparin was paused for an hour, but you assumed it kept going because
they only documented the rate every 2 hours. It made it impossible to
determine a start and end time without making some assumptions that would
alter the data, which is against our approach to releasing MIMIC (believe
me I did try to make the data more consistent).

Your best bet is to look at the series of vasopressor queries I wrote in
the MIMIC-III durations folder. The dose queries merge the dosing of
vasopressor across CareVue / MetaVision, which you could effectively adapt
to merge the doses across MIMIC-III / MIMIC-IV. The general pattern is that
they infer a start/stop time based on the rates documented in inputevents,
and accounting for some common quirks (e.g. nurses would set rates to 0 for
many hours if they wanted to stop the medication, but avoid having to
re-enter a lot of information when the medication was restarted). It should
be adaptable for heparin. If you get it to work you are welcome to put in a
PR!

**Reply by heisenbug-1** [reply]
Interesting :) In my case, there is no amount for the IV Drip(continuous) events in the CareVue dataset, only rate. The documentation says "For rates, the CHARTTIME represents the time at which that rate was set",  so I calculated the amount the patients got at each charttime, since the rate is hourly in case of heparin. I'm not using the MetaVision parts of MIMIC-III, since I only need patients from MIMIC-III, that are not in MIMIC-IV. I'm using MIMIC-IV for training, and MIMIC-III for predicting, so I need to keep the datasets distinct.

And you're 100% right, some events are not charted hourly. So if a patient received 1000UI oh heparin at 13:00, and the next charttime is 16:00 with the same rate, I assume the patient received 1000IU hourly from 1PM to 4PM, which is not 100% true in all cases, but I don't think I have enough data about the infusions to interpolate from :)

 I'll try to build something and create a PR, thank you for the feedback!

**Reply by alistairewj** [reply]
Yep, I think that's fair. One clarification for my post is I said MetaVision because all of MIMIC-IV had MetaVision as the ICU information system, so my mental model has MIMIC-III : CareVue // MIMIC-IV : MetaVision (for the MIMIC-III CareVue subset anyway).

Good luck!
